protocols/bftcosi/bftcosi.go

```diff
@@ -228,125 +240,94 @@ func (bft *ProtocolBFTCoSi) RegisterOnSignatureDone(fn func(*BFTSignature)) {
 	bft.onSignatureDone = fn
 }
 
-// Shutdown will close the dispatch-method if the protocol is stopped
-// before the normal termination.
+// Shutdown closes all channels in case we're done
 func (bft *ProtocolBFTCoSi) Shutdown() error {
-	bft.doneProcessing <- true
+	defer func() {
+		// In case the channels were already closed
+		recover()
```

Comment:
Here we only close the channels. The `panic` will come for the channels that have been closed in `Dispatch` (look at the new comment there).